lib/src/breakpoints_utils.dart

```diff
@@ -1,40 +1,65 @@
-import 'package:adaptive_breakpoints/adaptive_breakpoints.dart';
-
 import 'package:flutter/material.dart';
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
 
-extension Breakpoints on BuildContext {
-  /// All the breakpoints according to Material Design's breakpoint system.
-  BreakpointSystemEntry get breakpoints => getBreakpointEntry(this);
-
-  /// Double the default gutter size.
-  double get gutterLarge => (_material3Value ?? breakpoints.gutter) * 2;
-
-  /// The gutter size according to Material Design's breakpoints system.
-  double get gutter => _material3Value ?? breakpoints.gutter;
-
-  /// Half the default gutter size.
-  double get gutterSmall => (_material3Value ?? breakpoints.gutter) / 2;
-
-  /// A quarter the default gutter size.
-  double get gutterTiny => (_material3Value ?? breakpoints.gutter) / 4;
+import '../flutter_gutter.dart';
 
+/// A gap of the standard gutter size according to Material Design's breakpoints
+extension BreakpointExtensions on BuildContext {
   /// The margin size according to Material Design's breakpoints system.
-  double get margin => _material3Value ?? breakpoints.margin;
-
-  int get columns => breakpoints.columns;
+  double get margin => materialSpacing;
 
-  double? get _material3Value {
-    if (!Theme.of(this).useMaterial3) {
-      return null;
+  /// The size according to Material Design's with a specified [GutterType].
+  double gutter({GutterType type = GutterType.medium, double? scaleFactor}) {
+    switch (type) {
+      case GutterType.tiny:
+        return materialSpacing / (scaleFactor ?? 4);
+      case GutterType.small:
+        return materialSpacing / (scaleFactor ?? 2);
+      case GutterType.medium:
+        return materialSpacing;
+      case GutterType.large:
+        return materialSpacing * (scaleFactor ?? 2);
+      case GutterType.extraLarge:
```

Comment:
Shouldn't `scaleFactor` be a multiple on the entire switch statement?

If I had a GutterType of extra large and a scalefactor of 2, I'd expect the total scale factor to be 8.
Also looks like a bug where for small and tiny you're dividing by the scale factor instead of multiplying.

Reply:
Aah right i get what you mean, will fix.